### ISSUE TITLE
Gmail no longer work with cakephp

### ISSUE THREAD

**Issue by Asgraf** [cakephp/cakephp#7825]
````PHP
class EmailConfig {
    public $gmail = array(
        'host' => 'ssl://smtp.gmail.com',
        'port' => 465,
        'username' => 'my@gmail.com',
        'password' => 'secret',
        'transport' => 'Smtp'
    );
}
````
This causes following error:
````
stream_socket_client(): Peer certificate CN=`smtp.gmail.com' did not match expected CN=`ssl://smtp.gmail.com' stream_socket_client(): Failed to enable crypto stream_socket_client(): unable to connect to ssl://smtp.gmail.com:465 (Unknown error)
````
Tested on CakePHP v2.7.7 & PHP 5.6.16

**Comment by dereuromark** [comment]
I got the same issue yesterday with amazon and ssl.
Got it fixed for me by going back to tls=>true and possibly port 25.
Didn't have time to look into it more yesterday.

**Comment by dereuromark** [comment]
Looks like this could be responsible for the regression: https://github.com/cakephp/cakephp/pull/7766

**Comment by dereuromark** [comment]
Actually, it looks more like that could have been the same or a similar regression introduced in 2.7.7.
Versions 2.7.5 etc still seem to work.

**Comment by wiserfirst** [comment]
I got the same error and that's why I fixed it with PR #7766, so I think this can be closed.

Just need to wait for the next release. Until then try to use the latest code from Github

**Comment by markstory** [comment]
Closing as this has already been fixed on 2.7, and a release is coming pretty soon.

**Comment by dereuromark** [comment]
Also see https://github.com/cakephp/cakephp/pull/7834
